src/eip191_cosmwasm_client.ts

```diff
@@ -0,0 +1,973 @@
+import {
+  EncodeObject,
+  encodePubkey,
+  isOfflineDirectSigner,
+  makeAuthInfoBytes,
+  makeSignDoc,
+  OfflineDirectSigner,
+  OfflineSigner,
+  Registry,
+  TxBodyEncodeObject,
+} from '@cosmjs/proto-signing';
+import {
+  CometClient,
+  connectComet,
+  HttpEndpoint,
+} from '@cosmjs/tendermint-rpc';
+import { assert, assertDefined } from '@cosmjs/utils';
+import { SignMode } from 'cosmjs-types/cosmos/tx/signing/v1beta1/signing';
+import { TxRaw } from 'cosmjs-types/cosmos/tx/v1beta1/tx';
+import { ChangeAdminResult } from '@cosmjs/cosmwasm-stargate/build/signingcosmwasmclient';
+
+import {
+  AminoTypes,
+  calculateFee,
+  defaultRegistryTypes,
+  DeliverTxResponse,
+  GasPrice,
+  Event,
+  isDeliverTxFailure,
+  MsgSendEncodeObject,
+  SigningStargateClientOptions,
+  logs,
+  Attribute,
+  createProtobufRpcClient,
+  ProtobufRpcClient,
+  MsgDelegateEncodeObject,
+  MsgUndelegateEncodeObject,
+  MsgWithdrawDelegatorRewardEncodeObject,
+} from '@cosmjs/stargate';
+import {
+  CosmWasmClient,
+  ExecuteInstruction,
+  ExecuteResult,
+  InstantiateOptions,
+  InstantiateResult,
+  JsonObject,
+  MigrateResult,
+  MsgClearAdminEncodeObject,
+  MsgExecuteContractEncodeObject,
+  MsgInstantiateContract2EncodeObject,
+  MsgInstantiateContractEncodeObject,
+  MsgMigrateContractEncodeObject,
+  MsgStoreCodeEncodeObject,
+  MsgUpdateAdminEncodeObject,
+  UploadResult,
+} from '@cosmjs/cosmwasm-stargate';
+import {
+  MsgClearAdmin,
+  MsgExecuteContract,
+  MsgInstantiateContract,
+  MsgInstantiateContract2,
+  MsgMigrateContract,
+  MsgStoreCode,
+  MsgUpdateAdmin,
+} from 'cosmjs-types/cosmwasm/wasm/v1/tx';
+import { AccessConfig } from 'cosmjs-types/cosmwasm/wasm/v1/types';
+import { Coin } from 'cosmjs-types/cosmos/base/v1beta1/coin';
+import { BaseAccount } from 'cosmjs-types/cosmos/auth/v1beta1/auth';
+import {
+  MsgDelegate,
+  MsgUndelegate,
+} from '@neutron-org/neutronjs/cosmos/staking/v1beta1/tx';
+import { MsgWithdrawDelegatorReward } from 'cosmjs-types/cosmos/distribution/v1beta1/tx';
+import { Eip191Signer, isEip191Signer } from './eip191_signer';
+import { aminoConverters } from './amino';
+
+/**
+ * Signing information for a single signer that is not included in the transaction.
+ *
+ * @see https://github.com/cosmos/cosmos-sdk/blob/v0.42.2/x/auth/signing/sign_mode_handler.go#L23-L37
+ */
+export interface SignerData {
+  readonly accountNumber: number;
+  readonly sequence: number;
+  readonly chainId: string;
+}
+
+// Eip191SigningCosmwasmClient has the same interface as SigningCosmWasmClient from cosmjs.
+// It is mostly a copypaste of one with modified sign method that allows
+// for SIGN_MODE_EIP_191 to be used with the Neutron chain.
+export class Eip191SigningCosmwasmClient extends CosmWasmClient {
+  public readonly registry: Registry;
+  public readonly broadcastTimeoutMs: number | undefined;
+  public readonly broadcastPollIntervalMs: number | undefined;
+  private readonly signer: OfflineSigner | Eip191Signer;
```

Comment:
what about that idea about abstracted signer with adapters for each defined signer type that we talked about?